mlflow/pyfunc/scoring_server/__init__.py

```diff
@@ -401,6 +392,27 @@ def invocations(data, content_type, model, input_schema):
     return InvocationsResponse(response=result.getvalue(), status=200, mimetype="application/json")
 
 
+def _should_parse_as_unified_llm_input(json_input: dict):
+    return any(x in json_input for x in SUPPORTED_LLM_FORMATS)
+
+
+def _parse_json_data(data, metadata, input_schema):
+    json_input = _decode_json_input(data)
+    should_parse_as_unified_llm_input = _should_parse_as_unified_llm_input(json_input)
+    if should_parse_as_unified_llm_input:
+        # Unified LLM input format
+        if hasattr(metadata, "get_params_schema"):
+            params_schema = metadata.get_params_schema()
+        else:
+            params_schema = None
+        data, params = _split_data_and_params_for_llm_input(json_input, params_schema)
+    else:
+        # Traditional json input format
+        data, params = _split_data_and_params(data)
+        data = infer_and_parse_data(data, input_schema)
+    return data, params, should_parse_as_unified_llm_input
```

Comment:
I'd define a `dataclass` or `namedtuple`. A plain tuple, which requires indexing or unpacking, is a bit inconvenient.